run-cgprna/run_cgprna/fusion_tools.py

```diff
@@ -0,0 +1,173 @@
+                sys.exit('Error: Can not find second mate file of: %s' % a_mate)
+            if value == 2:
+                sys.exit('Error: Can not find first mate file of: %s' % a_mate)
+            sys.exit('Error: Internal code logic error')
+
+
+
+# NOTE: So far only used if Defuse is given a gzipped fastq file
+def gunzip(source_file, dest_file):
+    print('unzipping %s ...' % os.path.basename(source_file), flush=True)
+    with gzip.open(source_file, 'rb') as s_file, open(dest_file, 'wb') as d_file:
+        shutil.copyfileobj(s_file, d_file, 65536)
+    print('done.', flush=True)
+
+
+def run_fusion_wrapper(args, temp_dir_name, fusion_templates, no_gzip=False):
+    # args to dict to allow updates later
+    args_dict = copy.deepcopy(vars(args))
+    # only use temp_dir when needed to extract reference files
+    temp_dir = os.path.join(os.path.abspath(args.out_dir), temp_dir_name)
+    mkdir(temp_dir)
+
+    # valide inputs before bamtofastq, otherwise it could be to late
+    validate_input_seq_files(args.input)
+
+    # prepare the output dir
+    mkdir(args.out_dir)
+
+    reference_data_root=os.path.abspath(args.ref)
+
+    # Anything not a file will be treated as a reference root folder
+    if not os.path.isfile(reference_data_root):
+        if not os.path.exists(reference_data_root):
+            sys.exit('Error: cound not locate directory: %s' % reference_data_root)
+        if any(args_dict[ele] is None for ele in REF_RELATED_DEFAULTS.keys()):
+            sys.exit(
+                'Error: missing required input. When "--reference" is not a reference bundle tar file, you have to provide: %s' % ', '.join(
+                    [ '--' + key.replace('_', '-') for key in REF_RELATED_DEFAULTS.keys() if args_dict[key] is None])
+            )
+    elif not os.path.basename(reference_data_root).endswith('.tar.gz'):
+        # check if input ref file has valid file extensions
+        sys.exit('Error: wrong input format. "--reference" can only be a tar.gz file or a folder.')
+    else:
+        reference_data_root=os.path.join(temp_dir, 'ref')
+        # set ref related args to defaults if not given
+        for arg_name in REF_RELATED_DEFAULTS.keys():
+            if args_dict[arg_name] is None:
+                print('Set "%s" to default.' % arg_name)
+                args_dict[arg_name] = REF_RELATED_DEFAULTS[arg_name]
+            if arg_name == 'gene_build':
+                print('Make sure a folder named "%s" exists in the ref bundle.' % args_dict[arg_name])
+        untar(args.ref, os.path.join(reference_data_root, args_dict['species'], args_dict['ref_build']))
+
+    input_fastqs = []
+    fq_lane_name_count = 1
+
+    # dumb Defuse perl wrapper doesn't like gzipped, so: 
+    fq_suffix, bam_to_fq_template = ('fq.gz', BAM_TO_FASTQ) if not no_gzip else ('fq', BAM_TO_FASTQ_FOR_DEFUSE)
+
+    for a_raw_file in args.input:
+        # if input is not a bam, assume it's a fastq. i.e not supporing CRAM.
```

Comment:
Maybe a check for `.cram` returning a 'not supported' message could be helpful while debugging a failed run.